sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/DataLakeExceptionMappingPolicy.java

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.file.datalake.models;
+
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.storage.blob.models.BlobStorageException;
+import reactor.core.publisher.Mono;
+
+/**
+ * This is an exception mapping policy in an HttpPipeline to map BlobStorageExceptions to DataLakeStorageExceptions.
+ */
+public class DataLakeExceptionMappingPolicy implements HttpPipelinePolicy {
+
+    @Override
+    public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        return next.process().onErrorMap(DataLakeExceptionMappingPolicy::transformBlobStorageException);
```

Comment:
Could this call into `DataLakeImplUtils.transformBlobStorageException` instead of creating its own implementation of the method.

Reply:
I need to remove this class. I dont really use it anywhere.

Reply:
I tried to make a policy but it doesnt work because of the way we handle exceptions